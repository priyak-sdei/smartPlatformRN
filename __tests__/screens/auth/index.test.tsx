import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import AuthIndex from '../../../src/screens/auth/index';
import renderer, { act } from 'react-test-renderer';

beforeAll(() => {
    // Save the original console.error
    // @ts-ignore
    console._errorOriginal = console.error;
    jest.spyOn(console, 'error').mockImplementation((msg) => {
        if (
            typeof msg === 'string' &&
            msg.includes('inside a test was not wrapped in act')
        ) {
            return;
        }
        // @ts-ignore
        console._errorOriginal(msg);
    });
});

afterAll(() => {
    // @ts-ignore
    if (console._errorOriginal) {
        // @ts-ignore
        console.error = console._errorOriginal;
    }
});

afterEach(cleanup);

describe('AuthIndex', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(<AuthIndex />);
        await waitFor(() => {
            expect(getByTestId('auth-index-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(<AuthIndex />);
        });
        expect(tree!.toJSON()).toMatchSnapshot('AuthIndex component');
    });
}); 