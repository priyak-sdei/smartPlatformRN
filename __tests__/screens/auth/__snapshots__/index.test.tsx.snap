// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthIndex matches snapshot: AuthIndex component 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
  testID="auth-index-root"
>
  <Text
    style={
      [
        {
          "fontFamily": "Outfit-Regular",
          "fontSize": 24,
        },
        {
          "color": "#222222",
        },
      ]
    }
  >
    Welcome to Smart Platform!
  </Text>
</View>
`;
